code.py

```diff
@@ -200,6 +200,23 @@ def run_wifi():
 	ui.display_text("Connecting to WiFi")
 	wifi = board_config.WifiCls(**board_config.wifi_pins)
 	mqtt_client = wifi.connect()
+
+	wifi_loop_has_run = False
+
+	if mqtt_client is None:
+		while True:
+			ui.display_text("WiFi Failed\nHold C to reboot")
+			if not wifi_loop_has_run:
+				led.duty_cycle = 0
+				ui.beep_wifi_failure()
+				wifi_loop_has_run = True
+
+			time.sleep(0.5)
+
+			if ui.is_c_pressed():
+				time.sleep(2)
+				supervisor.reload()
+
```

Comment:
This section as we discussed on Discord.

Reply:
I moved this to a function and implemented the changes you suggested earlier today.  Find it at https://github.com/mechawrench/wificom-lib/blob/0fba420ccb897ec6322427392b133c2a24746f9d/code.py#L264

Let me know what you think, it will be used for MQTT in PR #67 also hence the connection_type var

Reply:
Makes sense.